### PR TITLE
Support compiling for BMSLD

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

File: src/mercury_engine_data_structures/game_check.py [hunk 1]
```diff
@@ -102,6 +102,18 @@ def is_sr_or_else(subcon1, subcon2) -> IfThenElse:
     return IfThenElse(construct.this._params.target_game == Game.SAMUS_RETURNS.value, subcon1, subcon2)
 
 
+def is_at_most(target: Game, subcon1, subcon2) -> IfThenElse:
```
Comment (Miepee) [inline src/mercury_engine_data_structures/game_check.py:105]: when should this be used over `IfThenElse(
    game_check.current_game_at_most(Game.SAMUS_RETURNS),
    subcon1,
    subcon2,
)`?

Comment (Miepee) [inline src/mercury_engine_data_structures/game_check.py:105]: I prefer the terms `subcontrue` and `subconfalse` just to make it clearer when each subcon is getting used.

**Review by Miepee — COMMENTED** [review]
tests are failing

File: src/mercury_engine_data_structures/game_check.py (hunk 1 above)
Comment (henriquegemignani) [reply src/mercury_engine_data_structures/game_check.py:105]: Always. I want to remove `game_check.current_game_at_most` later on as it's not safe with compile.

**Review by Miepee — COMMENTED** [review]
Changes look fine to me, but your tests are failing still.

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**